### PR TITLE
Upgrade to ruby 2.3.0 and activerecord 4.2.6

### PR DESCRIPTION
**Pull request by h-lame**
ActiveRecord and Ruby are the main things we care about (as per spreadsheet) but we also bump most of our dependencies while we're at it.

This is the sibling PR to https://github.com/alphagov/transition/pull/508 which does similar upgrades to transition. These two apps share a db and it's better if their versions of AR are similar to avoid calamity.

### CODE REVIEW

File: spec/features/http_request_handling_spec.rb [hunk 1]
```diff
@@ -37,35 +37,61 @@
...
+    describe '#status' do
+      subject { super().status }
+      it { is_expected.to eq(500) }
+    end
+
+    describe '#location' do
+      subject { super().location }
+      it { is_expected.to eq(nil) }
+    end
+    specify         { expect(last_response.headers).not_to include('Cache-Control') }
```
Comment (davidbasalla) [inline spec/features/http_request_handling_spec.rb]: niggle alert - can the empty whitespace after `specify` be removed? Looks like it was trying to line up the first '{' with the previous brackets when using `its`

**Comment by davidbasalla** [comment]
Would it have been an option to use https://github.com/rspec/rspec-its to deal with `its`? Or would that be less advisable as it adds another gem dependency to the mix (which by the looks of it hasn't been updated in a year as well).

File: spec/features/http_request_handling_spec.rb (hunk 1 above)
Comment (h-lame) [inline spec/features/http_request_handling_spec.rb]: Good spot - will do.

**Comment by h-lame** [comment]
@davidbasalla yeah - we could have used rspec-its but as something that's been pulled out of rspec and doesn't look like it's being updated, I felt it wasn't worth the future hassle especially as transpec can convert it away right now, but there's no guarantee it'll always be able to for future rspec syntax changes.

File: spec/features/http_request_handling_spec.rb (hunk 1 above)
Comment (h-lame) [inline spec/features/http_request_handling_spec.rb]: Rebased away.  Thanks!

**Comment by davidbasalla** [comment]
Looks good to me! :+1:

**Pull request merged**